contracts/src/PDPService.sol

```diff
@@ -308,12 +347,45 @@ contract PDPService {
         return result;
     }
 
-
     struct Proof {
         bytes32 leaf;
         bytes32[] proof;
     }
 
+    // Roll over to the next proving period
+    // 1. Resample challenge
+    // 2. Track newly added roots for proving
+    // 3. Remove scheduled removals
+    function nextProvingPeriod(uint256 setId) public {
+        require(msg.sender == proofSetOwner[setId], "only the owner can move to next proving period");
+        // Take removed roots out of proving set
+        uint256[] storage removals = scheduledRemovals[setId];
+        uint256[] memory removalsToProcess = new uint256[](removals.length);
+    
+        for (uint256 i = 0; i < removalsToProcess.length; i++) {
```

Comment:
Should there be some limit on how many removals can be processed at once? If there isn't a limit it looks like it is possible to brick a proving set with too many removals

Reply:
Yeah out of concern for this I put this check here: https://github.com/FILCAT/pdp/pull/46/files#diff-712b2db21b090e745391534efb9ce9dfb70c3f729952e9433b627e264e3f83a3R261.  One of the things I'm most worried about with this change is how to set this number.  I arbitrarily set it to 2000 after eyeballing batch removal gas costs.  But its possible this could be either annoyingly low or dangerously high.